Dockerfile

```diff
@@ -1,19 +1,28 @@
-FROM alpine
+FROM debian:11 as builder
+ARG PDNS_VERSION=4.7.3
+
+WORKDIR /build
+RUN apt update && \
+    apt install -y curl bzip2 g++ python3-venv libtool make pkg-config \
+    libboost-all-dev libssl-dev libluajit-5.1-dev libcurl4-openssl-dev libsqlite3-dev
+RUN curl -sL https://downloads.powerdns.com/releases/pdns-$PDNS_VERSION.tar.bz2 | tar -jx
+WORKDIR /build/pdns-$PDNS_VERSION
+RUN ./configure --with-modules='bind gsqlite3' && \
+    make -j $(nproc) && \
+    make install
+
+FROM debian:11-slim
+
+RUN apt update && apt install -y curl sqlite3 luajit && apt clean
 
 ARG BOOTSTRAP_SQL=schema.sqlite3-4.5.x.sql
 
 ADD entrypoint.sh /entrypoint/script
 ADD $BOOTSTRAP_SQL /entrypoint/bootstrap.sql
 
-RUN apk add --no-cache \
-    pdns \
-    pdns-backend-sqlite3 \
-    sqlite \
-    pdns-tools
-
-ENV VERSION=4.5
+COPY --from=builder /usr/local /usr/local
 
 EXPOSE 53 53/udp
 
-ENTRYPOINT [ "/bin/sh", "/entrypoint/script" ]
-CMD [ "/usr/sbin/pdns_server" ]
+ENTRYPOINT [ "/bin/bash", "/entrypoint/script" ]
```

Comment:
Debian's `/bin/sh` complains that `pipefail` is not a valid option.

Reply:
It is in fact no portable shell but bash shell so I see the point on changing it. Thanks!